### ISSUE TITLE
Bug #1860 needs to be re-opened

### ISSUE THREAD

**Issue by g2graman** [balderdashy/sails#2736]
As the title reads; I can't find a single source that seems properly lift an app in a testing environment without complaining about maxAge being undefined upon lifting:

    1)  "before all" hook:
    Uncaught TypeError: Cannot read property 'maxAge' of undefined

I posted the last comment there (https://github.com/balderdashy/sails/issues/1860#issuecomment-77801626) but it might not get seen because the initial issue has been closed.

**Comment by Winangel** [comment]
I confirm this issue. Got the same error on Sails.Lift() in my bootstrap test file.

**Comment by mikemastah** [comment]
I am not sure what exactly you are trying to accomplish. Do you happen to have a more detailed error message? 
But to me it seems that the following:
```
config:{
    cache: {
        maxAge: 900000
    }
},
```
Should actually be
```
cache: {
    maxAge: 900000
},
```

**Comment by g2graman** [comment]
My purpose was to get the app to lift before a barrage of unit tests by mimicking the settings from before.

That seems to have fixed the initial problem. but now there's a new one:
```
  1) "before all" hook

  0 passing (2s)
  1 failing

  1)  "before all" hook:
     Uncaught Error: static() root path required
      at Function.module.exports (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/express/node_modules/connect/lib/middleware/static.js:49:20)
      at _.defaults.www (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/http/middleware/defaults.js:26:49)
      at module.exports (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/http/middleware/defaults.js:36:7)
      at loadExpress (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/http/initialize.js:103:65)
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/http/index.js:190:18
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/app/private/after.js:91:14
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:232:13
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:119:25
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:24:16
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:229:17
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:516:34
      at handlerFn (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/app/private/after.js:78:13)
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:511:21
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:227:13
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:111:13
      at Array.forEach (native)
      at _each (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:32:24)
      at async.each (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:110:9)
      at _asyncMap (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:226:9)
      at Object.map (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:204:23)
      at _parallel (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:509:20)
      at Object.async.parallel (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:539:9)
      at Sails.emitter.after (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/app/private/after.js:89:11)
      at Hook.initialize (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/http/index.js:189:15)
      at Hook.bound [as initialize] (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/lodash/dist/lodash.js:729:21)
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/lib/hooks/index.js:132:16
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:425:17
      at /home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:419:17
      at Array.forEach (native)
      at _each (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:32:24)
      at Object.taskComplete (/home/g2graman/Desktop/UofTHacks2015/node_modules/sails/node_modules/async/lib/async.js:418:13)
      at processImmediate [as _immediateCallback] (timers.js:330:15)

```

**Comment by tjwebb** [comment]
Any update on this?

**Comment by mikemastah** [comment]
Looking at [this line](https://github.com/balderdashy/sails/blob/master/lib/hooks/http/middleware/defaults.js#L26), I'd say you have to explicitly set your static path by adding something like..
```(json)
paths: {
    public: '/path/to/static/folder/.tmp'
}
```
..to your config object.  Correct me if I am wrong, but this does not appear to be documented anywhere

**Comment by g2graman** [comment]
@tjwebb @mikemastah Have not been able to fix the issue yet and mike's suggestion didn't seem to do the trick, I still get `Uncaught TypeError: Cannot read property 'maxAge' of undefined` so I've returned my `bootstrap.test.js` to a bare state:

    var Sails = require('sails'), sails;

    before(function(done) {
      Sails.lift({
        // configuration for testing purposes
      }, function(err, server) {
        sails = server;
        if (err) return done(err);
        // here you can load fixtures, etc.
        done(err, sails);
      });
    });

    after(function(done) {
      // here you can clear fixtures, etc.
      sails.lower(done);
   });

Any suggestions on making it so I can successfully lift the app before all tests?

**Comment by mikemastah** [comment]
That does look pretty basic. I have tried multiple times and I am not able to reproduce your error..
I have uploaded a super basic sails.js test setup to my github repo [sails-test-example](/mikemastah/sails-test-example). Could you maybe clone that and try to run `npm test`  (After running `npm install` of course). That should definitely work, and maybe you can figure out what to change in your code to make it work.

Let me know if that works out!

**Comment by CWyrtzen** [comment]
any resolution here?

**Comment by sailsbot** [comment]
Thanks for posting, @g2graman.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
